src/cr/cube/measures/pairwise_significance.py

```diff
@@ -4,11 +4,125 @@
+            t_stats = self._t_stats[:, i, :]
+            p_vals = self._p_vals[:, i, :]
+            values.append(OverlapSignificance(t_stats, p_vals))
+        return values
+
+    @lazyproperty
+    def _t_stats(self):
+        return np.array(
+            [
+                PerSubvariableOverlapsSignificance(
+                    self._slice.overlaps[i], self._slice.valid_overlaps[i]
+                ).t_stats
+                for i in range(len(self._slice.row_labels))
+            ]
+        )
+
+    @lazyproperty
+    def _p_vals(self):
+        return np.array(
+            [
+                PerSubvariableOverlapsSignificance(
+                    self._slice.overlaps[i], self._slice.valid_overlaps[i]
+                ).p_vals
+                for i in range(len(self._slice.row_labels))
+            ]
+        )
+
+
+class PerSubvariableOverlapsSignificance(object):
+    def __init__(self, overlaps, valid_overlaps):
+        self._overlaps = overlaps
+        self._valid_overlaps = valid_overlaps
+
+    @lazyproperty
+    def n_subvars(self):
+        return self._overlaps.shape[0]
+
+    @lazyproperty
+    def values(self):
+        return [
+            _SubvarPairwiseSignificance(
+                self._overlaps, self._valid_overlaps, subvar_idx
+            )
+            for subvar_idx in range(self.n_subvars)
+        ]
+
+    @lazyproperty
+    def t_stats(self):
+        return np.array([svps.t_stats for svps in self.values])
+
+    @lazyproperty
+    def p_vals(self):
+        return np.array([svps.p_vals for svps in self.values])
+
+
+class _SubvarPairwiseSignificance(object):
+    def __init__(self, overlaps, valid_overlaps, subvar_idx):
+        self._overlaps = overlaps
+        self._valid_overlaps = valid_overlaps
+        self._subvar_idx = subvar_idx
+
+    @lazyproperty
+    def n_subvars(self):
+        return self._overlaps.shape[0]
+
+    @lazyproperty
+    def t_stats_and_pvals(self):
+        idx, vo, so = (self._subvar_idx, self._valid_overlaps, self._overlaps)
+        Na = vo[idx, idx]
+        Sa = so[idx, idx]
+        pa = Sa / Na
+        z_scores = []
+        p_vals = []
+        for i in range(self.n_subvars):
+            if i == idx:
+                z_scores.append(0)
+                p_vals.append(0)
+                continue
+            Nb = vo[i, i]
+            Nab = vo[i, idx]
+            Sb = so[i, i]
+            Sab = so[i, idx]
+
+            pb = Sb / Nb
+            pab = Sab / Nab
+            N = Na + Nb - Nab
+            zab = (pa - pb) / np.sqrt(
+                1 / N * (pa * (1 - pa) + pb * (1 - pb) + 2 * pa * pb - 2 * pab)
+            )
+            z_scores.append(zab)
+            pab = 2 * (1 - t.cdf(abs(zab), df=N - 2))
+            p_vals.append(pab)
+        return np.array(z_scores), np.array(p_vals)
+
+    @lazyproperty
+    def t_stats(self):
+        return self.t_stats_and_pvals[0]
+
+    @lazyproperty
+    def p_vals(self):
+        return self.t_stats_and_pvals[1]
```

Comment:
before getting into this i'd like to see some comments/doctrings ... it's hard to understand what's going on :D sorry